src/main/java/org/testng/internal/XmlMethodSelector.java

```diff
@@ -231,11 +187,19 @@ private boolean hasExcludedMethods() {
     return vResult;
   }
 
-  private String makeMethodName(String className, String methodName) {
+  private static String methodName(String methodName) {
```

Comment:
@juherr - I don't quite understand this condition part of ensuring that `$` is not there, but checking if `\\$` is there. For e.g., lets say I have an input of `random\\$`, we would always evaluate to true and the method name would be returned as is, so I am guessing that the `replaceAll()` call never happens right ?

Reply:
@krmahadevan Any idea?

Reply:
My apologies for missing this one @juherr .. I didn't see this at all. I will go home and take another look at this and get back if I find anything...

Reply:
@juherr - We can do something like this 

```java
private static String methodName(String methodName) {
    if (methodName.contains("\\$")) {
        return methodName;
    }
    return methodName.replaceAll("\\Q$\\E", Matcher.quoteReplacement("\\$"));
}
```

Reply:
I like this option! :+1: